lib/coverage_var.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @fileOverview This file is used to determine the coverage variable name
+ * for a module. The coverage variable holds all the coverage information of 
+ * the module "under test". This variable exists in the instrumented version
+ * of the module. In order to generate an instrumented version, use istanbul instrument.
+ * 
+ * We've followed the exact same steps instanbul follows to generate the names.
+ * https://github.com/gotwarlost/istanbul/blob/master/lib/instrumenter.js#L43
+ *
+ **/
+
+const { Cc, Ci, Cu } = require("chrome");
+
+/**
+ * @namespace Helper object to generate a md5 hash for a given input string.
+ */
+var crypto = {
+
+  /**
+   * Returns a base64 md5sum hash of a given input string.
+   *
+   * @param {string} str Input string
+   */
+  hash: function(str) {
+    var converter = Cc["@mozilla.org/intl/scriptableunicodeconverter"].
+                      createInstance(Ci.nsIScriptableUnicodeConverter);
+    converter.charset = "UTF-8";
+    // result is an out parameter,
+    // result.value will contain the array length
+    var result = {};
+    // data is an array of bytes
+    var data = converter.convertToByteArray(str, result);
+    var ch = Cc["@mozilla.org/security/hash;1"].
+               createInstance(Ci.nsICryptoHash);
+    ch.init(ch.MD5);
+    ch.update(data, data.length);
+    var hash = ch.finish(false);
+    return crypto.base64(hash);
+  },
+  
+  /**
+   * Returns a base64 version of a given input hash.
+   *
+   * @param {string} input Input
+   * @returns {string} output base64 version of the input.
+   */
+  base64: function(input) {
```

Comment:
I've picked this as it is from [here](https://github.com/Gozala/crypto/blob/master/md5.js#L90). I'm assuming I can do that under BSD license.